MapboxNavigation/NavigationMapView.swift

```diff
@@ -442,39 +469,139 @@ open class NavigationMapView: MGLMapView, UIGestureRecognizerDelegate {
+        mainRouteLayer.lineWidth = NSExpression(format: "mgl_interpolate:withCurveType:parameters:stops:($zoomLevel, 'linear', nil, %@)", MBRouteLineWidthByZoomLevel)
+        mainRouteLayer.lineJoin = NSExpression(forConstantValue: "round")
+
+        if routeGradientStops.line.isEmpty == false {
+            mainRouteLayer.lineGradient = NSExpression(format: "mgl_interpolate:withCurveType:parameters:stops:($lineProgress, 'linear', nil, %@)", routeGradientStops.line)
+        }
+
+        return mainRouteLayer
+    }
+
+    func mainRouteCasingStyleLayer(identifier: String, source: MGLSource) -> MGLLineStyleLayer {
+        let mainRouteCasingLayer = MGLLineStyleLayer(identifier: identifier, source: source)
+        mainRouteCasingLayer.predicate = NSPredicate(format: "isAlternateRoute == false")
+        // Default color if no traffic is enabled
+        mainRouteCasingLayer.lineColor = NSExpression(forConstantValue: routeCasingColor)
+        mainRouteCasingLayer.lineWidth = NSExpression(format: "mgl_interpolate:withCurveType:parameters:stops:($zoomLevel, 'linear', nil, %@)", MBRouteLineWidthByZoomLevel.multiplied(by: 1.5))
+        mainRouteCasingLayer.lineJoin = NSExpression(forConstantValue: "round")
+
+        if routeGradientStops.casing.isEmpty {
+            mainRouteCasingLayer.lineGradient = NSExpression(format: "mgl_interpolate:withCurveType:parameters:stops:($lineProgress, 'linear', nil, %@)", routeGradientStops.casing)
+        }
+
+        return mainRouteCasingLayer
+    }
+
+    func alternateRouteStyleLayer(identifier: String, source: MGLSource) -> MGLLineStyleLayer {
+        let alternateRoutesLayer = MGLLineStyleLayer(identifier: identifier, source: source)
+        alternateRoutesLayer.predicate = NSPredicate(format: "isAlternateRoute == true")
+        alternateRoutesLayer.lineColor = NSExpression(forConstantValue: routeAlternateColor)
+        alternateRoutesLayer.lineWidth = NSExpression(format: "mgl_interpolate:withCurveType:parameters:stops:($zoomLevel, 'linear', nil, %@)", MBRouteLineWidthByZoomLevel)
+        alternateRoutesLayer.lineJoin = NSExpression(forConstantValue: "round")
+        return alternateRoutesLayer
+    }
+
+    func alternateRouteCasingStyleLayer(identifier: String, source: MGLSource) -> MGLLineStyleLayer {
+        let alternateRoutesCasingLayer = MGLLineStyleLayer(identifier: identifier, source: source)
+        alternateRoutesCasingLayer.predicate = NSPredicate(format: "isAlternateRoute == true")
+        alternateRoutesCasingLayer.lineColor = NSExpression(forConstantValue: routeAlternateCasingColor)
+        alternateRoutesCasingLayer.lineWidth = NSExpression(format: "mgl_interpolate:withCurveType:parameters:stops:($zoomLevel, 'linear', nil, %@)", MBRouteLineWidthByZoomLevel.multiplied(by: 1.5))
+        alternateRoutesCasingLayer.lineJoin = NSExpression(forConstantValue: "round")
+        return alternateRoutesCasingLayer
+    }
+
+    func fadeRoute(_ fractionTraveled: Double) {
```

Comment:
Maybe call it something more generic like `updateRouteLine(fractionTraveled:)` (with a named argument).